File: Backend/src/test/java/lan/dk/podcastserver/utils/MatcherExtractorTest.java
package lan.dk.podcastserver.utils;

import javaslang.control.Option;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Created by kevin on 04/03/2017
 */
public class MatcherExtractorTest {

    @Test
    public void should_return_value() {
        /* GIVEN */
        String s = "foo";
        String p = "(.*)";

        /* WHEN  */
        Option<String> v = MatcherExtractor.from(p).extractFrom(s).group(1);

        /* THEN  */
        assertThat(v).containsExactly("foo");
    }

    @Test
    public void should_return_no_value() {
        /* GIVEN */
        String s = "";
        String p = "abc";

        /* WHEN  */
        Option<String> v = MatcherExtractor.from(p).extractFrom(s).group(1);

        /* THEN  */
        assertThat(v).isEqualTo(Option.none());
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/utils/MatcherExtractorTest.kt
package com.github.davinkevin.podcastserver.utils

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.regex.Pattern

class MatcherExtractorTest {

    @Test
    fun `should extract value`() {
        /* GIVEN */
        val s = "foo"
        val p = Pattern.compile("(.*)")

        /* WHEN  */
        val v = MatcherExtractor.from(p).on(s).group(1)

        /* THEN  */
        assertThat(v).isEqualTo("foo")
    }

    @Test
    fun `should not return any value`() {
        /* GIVEN */
        val s = ""
        val p = Pattern.compile("abc")

        /* WHEN  */
        val v = MatcherExtractor.from(p).on(s).group(1)

        /* THEN  */
        assertThat(v).isNull()
    }
}
